src/pokeblock_feed.c

```diff
@@ -0,0 +1,1012 @@
+    {
+        gUnknown_03005F34 += sMonPokeblockAnims[animID][4];
+        if (sMonPokeblockAnims[animID][9] == 1)
+            break;
+    }
+}
+
+static void sub_814825C(void)
+{
+    switch (gUnknown_03005F3C)
+    {
+    case 0:
+        gUnknown_03005F40 = sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][0];
+        gPokeblockFeedPokeSprite = &gSprites[gPokeblockFeedMonSpriteID];
+        gPokeblockFeedPokeSpriteCopy = *gPokeblockFeedPokeSprite;
+        gUnknown_03005F3C = 10;
+        break;
+    case 1 ... 9:
+        break;
+    case 10:
+        sub_8148540();
+        if (sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1] != 0)
+        {
+            gPokeblockFeedPokeSprite->oam.affineMode = 3;
+            gPokeblockFeedPokeSprite->oam.matrixNum = 0;
+            gPokeblockFeedPokeSprite->affineAnims = sSpriteAffineAnimTable_8412050;
+            InitSpriteAffineAnim(gPokeblockFeedPokeSprite);
+        }
+        gUnknown_03005F3C = 50;
+    case 50:
+        if (sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1] != 0)
+        {
+            if (gPokeblockMonNotFlipped == 0)
+                StartSpriteAffineAnim(gPokeblockFeedPokeSprite, sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1] + 10);
+            else
+                StartSpriteAffineAnim(gPokeblockFeedPokeSprite, sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1]);
+        }
+        gUnknown_03005F3C = 60;
+        break;
+    case 60:
+        if (sub_81485CC() == 1)
+        {
+            if (gUnknown_03005FA0[9] == 0)
+            {
+                gUnknown_03005F40++;
+                sub_8148540();
+                gUnknown_03005F3C = 60;
+            }
+            else
+            {
+                FreeOamMatrix(gPokeblockFeedPokeSprite->oam.matrixNum);
+                gUnknown_03005F3C = 70;
+            }
+        }
+        break;
+    case 70:
+        FreePokeSpriteMatrix();
+        gUnknown_03005F40 = 0;
+        gUnknown_03005F3C = 0;
+        break;
+    case 71 ... 90:
+        break;
+    }
+}
+
+static bool8 sub_8148540(void)
+{
+    u8 i;
+    for (i = 0; i < 10; i++)
+        gUnknown_03005FA0[i] = sMonPokeblockAnims[gUnknown_03005F40][i];
+    if (gUnknown_03005FA0[4] == 0)
+        return 1;
+    else
+    {
+        gUnknown_03005FA0[10] = Sin(gUnknown_03005FA0[0], gUnknown_03005FA0[2]);
+        gUnknown_03005FA0[11] = Cos(gUnknown_03005FA0[0], gUnknown_03005FA0[3]);
+        gUnknown_03005FA0[12] = gUnknown_03005FA0[4];
+        gUnknown_03005FA0[13] = gPokeblockFeedPokeSprite->pos2.x;
+        gUnknown_03005FA0[14] = gPokeblockFeedPokeSprite->pos2.y;
+        sub_8148710();
+        gUnknown_03005FA0[4] = gUnknown_03005FA0[12];
+        sub_814862C();
+        gUnknown_03005FA0[4] = gUnknown_03005FA0[12];
+        return 0;
+    }
+}
+
+static bool8 sub_81485CC(void)
+{
+    u16 var = gUnknown_03005FA0[12] - gUnknown_03005FA0[4];
+
+    gPokeblockFeedPokeSprite->pos2.x = (*((u16*)(&ewram[0x1D000]) + var));
```

Comment:
Does it match if you do this?
`#define ewram1D000 ((u16 *)(ewram + 0x1D000))`
`gPokeblockFeedPokeSprite->pos2.x = ewram1D000[var];`